### PR TITLE
C++14 Workaround issue #349

### PR DESCRIPTION
**Pull request by crtrott**
Fixes C++14 build issues.

### CODE REVIEW

File: src/blas/impl/KokkosBlas3_gemm_impl.hpp [hunk 1]
```diff
@@ -46,6 +46,14 @@
 
 #include<Kokkos_Core.hpp>
 
+#ifdef KOKKOS_ENABLE_CXX14
```
Comment (mhoemmen) [inline src/blas/impl/KokkosBlas3_gemm_impl.hpp]: This is not a blocker, but one-liners are nice:
```c++
#if defined(KOKKOS_ENABLE_CXX14) && defined(KOKKOS_COMPILER_GNU) && KOKKOS_COMPILER_GNU <= 720
#  define KOKKOS_IMPL_BATCHED_GEMM_GCC_CXX14_WORKAROUND
#endif 
```

File: src/blas/impl/KokkosBlas3_gemm_impl.hpp [hunk 2]
```diff
@@ -77,16 +85,26 @@ struct impl_deep_copy_matrix_block<TeamHandle,ViewTypeScratch,ViewType,Layout,bl
   static void copy(const TeamHandle& team, const ViewTypeScratch& A_scr, const ViewType& A, const int& offset_i, const int& offset_j) {
     if(offset_i + blockDim_i <= A.extent_int(0) && offset_j + blockDim_j <= A.extent_int(1)) {
       Kokkos::parallel_for(Kokkos::TeamThreadRange(team,blockDim_j), [&] (const int j) {
+#ifndef KOKKOS_IMPL_BATCHED_GEMM_GCC_CXX14_WORKAROUND
         const int idx_j = offset_j+j;
+#endif
         Kokkos::parallel_for(Kokkos::ThreadVectorRange(team,blockDim_i), [&] (const int i) {
+#ifdef KOKKOS_IMPL_BATCHED_GEMM_GCC_CXX14_WORKAROUND
+          const int idx_j = offset_j+j;
+#endif
```
Comment (kyungjoo-kim) [inline src/blas/impl/KokkosBlas3_gemm_impl.hpp]: It probably necessary to pu ``int idx_j = offset_j + j`` with ``#else``.

Comment (kyungjoo-kim) [reply src/blas/impl/KokkosBlas3_gemm_impl.hpp]: Oh... sorry... I just saw that the ``idx_j`` is defined on the above.

**Comment by kyungjoo-kim** [comment]
It is okay with white. Plz, merge this PR.

**Comment by srajama1** [comment]
@kyungjoo-kim Thank you !

**Pull request merged**